src/backend/distributed/planner/distributed_planner.c

```diff
@@ -97,12 +98,49 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 	Query *originalQuery = NULL;
 	PlannerRestrictionContext *plannerRestrictionContext = NULL;
 	bool setPartitionedTablesInherited = false;
+	bool constraintCheckQuery = false;
 
 	if (cursorOptions & CURSOR_OPT_FORCE_DISTRIBUTED)
 	{
 		needsDistributedPlanning = true;
 	}
 
+	if (AlterTableInProgress() && parse->commandType == CMD_SELECT &&
```

Comment:
could we move this logic into a function maybe? It is in the `distributed_planner ()` and looks like too much information for readers of this function.

Reply:
Alternatively, it would be nice if we could handle this case in ExecutorRun somehow. We have a similarly nasty hack in ExecutorStart.

`distributed_planner` is the main entry point into Citus so it would be nice to keep it free of clutter.